### PR TITLE
Update `timeline.spec.ts` - use Cypress Testing Library

### PR DESCRIPTION
**Pull request by luixxiul**
Closes https://github.com/vector-im/element-web/issues/25033

type: task

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: cypress/e2e/timeline/timeline.spec.ts [hunk 1]
```diff
@@ -225,10 +226,11 @@ describe("Timeline", () => {
             );
 
             // Wait until configuration is finished
-            cy.contains(
-                ".mx_RoomView_body .mx_GenericEventListSummary .mx_GenericEventListSummary_summary",
-                "created and configured the room.",
-            ).should("exist");
+            cy.get(".mx_RoomView_body .mx_GenericEventListSummary[data-layout='group']").within(() => {
+                cy.get(".mx_GenericEventListSummary_summary")
```
Comment (weeman1337) [inline cypress/e2e/timeline/timeline.spec.ts:230]: I would personally remove `get(".mx_GenericEventListSummary_summary")` . This query does already run „within“ `.mx_GenericEventListSummary`.

→

```
cy.fndByText(OLD_NAME + " created and configured the room.")
```

File: cypress/e2e/timeline/timeline.spec.ts [hunk 2]
```diff
@@ -776,11 +788,13 @@ describe("Timeline", () => {
             cy.getComposer().type(`${MESSAGE}{enter}`);
 
             // Reply to the message
-            cy.contains(".mx_RoomView_body .mx_EventTile_line", "Hello world")
-                .realHover()
+            cy.get(".mx_EventTile_last")
                 .within(() => {
-                    cy.findByRole("button", { name: "Reply" }).click();
-                });
+                    cy.findByText("Hello world", { timeout: 1000 });
```
Comment (weeman1337) [inline cypress/e2e/timeline/timeline.spec.ts:793]: Can you remove the timeout? Sometimes tests run slowly. This is a possible cause for flaky tests.

**Review by weeman1337 — CHANGES_REQUESTED** [review]
:+1:  thank you for your contribution @luixxiul .

Looks good in general. Just a few minor comments.

**Comment by luixxiul** [comment]
@weeman1337 The PR should be fine. Would you please have a look at again when you have some time?

**Review by weeman1337 — APPROVED** [review]
/lgtm thank you @luixxiul  :+1:

**Pull request merged**